File: backend/pokerogue/src/main/java/com/pokerogue/helper/pokemon/dto/EvolutionResponse.java
package com.pokerogue.helper.pokemon.dto;


import com.pokerogue.helper.pokemon.config.ImageUrl;
import com.pokerogue.helper.pokemon.data.Evolution;
import com.pokerogue.helper.pokemon.data.Pokemon;

public record EvolutionResponse(
        String id,
        String name,
        Integer level,
        Integer depth,
        String item,
        String condition,
        String image
) {

    public static EvolutionResponse from(Pokemon pokemon, Evolution evolution, Integer depth) {
        return new EvolutionResponse(
                pokemon.getId(),
                pokemon.getKoName(),
                evolution.getLevel(),
                depth,
                evolution.getItem(),
                "",
                ImageUrl.getPokemonImage(pokemon.getImageId())
        );
    }
}

File: android/remote/src/main/java/poke/rogue/helper/remote/dto/response/pokemon/EvolutionResponse.kt
package poke.rogue.helper.remote.dto.response.pokemon

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class EvolutionResponse(
    @SerialName("id")
    val pokemonId: String = "",
    @SerialName("name")
    val pokemonName: String,
    @SerialName("image")
    val imageUrl: String,
    val depth: Int,
    val level: Int,
    val item: String? = null,
    val condition: String? = null,
)

@Serializable
data class EvolutionsResponse(
    val currentDepth: Int,
    @SerialName("stages")
    val evolutions: List<EvolutionResponse>,
)
